<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>959</width>
    <height>682</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="accessibleDescription">
    <string/>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>0</y>
      <width>291</width>
      <height>181</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="RadioButtons">
     <item>
      <widget class="QRadioButton" name="ButtonMP">
       <property name="text">
        <string>Алгоритм средней точки</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="ButtonCA">
       <property name="text">
        <string>Алгоритм на основе канонического уравнения</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="ButtonBR">
       <property name="text">
        <string>Алгоритм Брезенхема</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="ButtonPA">
       <property name="text">
        <string>Алгоритм на основе параметрического уравнения</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="ButtonBA">
       <property name="text">
        <string>Библиотечный алгоритм</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>0</y>
      <width>591</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_line">
       <property name="text">
        <string>Цвет рисования(текущий цвет-черный)</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonBlack">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(0,0,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonWhite">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(255,255,255);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonYellow">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(255,255,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonRed">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(255,0,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonOrange">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(255,104,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonPurple">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(128,0,128);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonBlue">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(0,0,255);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonGreen">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(0,255,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>40</y>
      <width>461</width>
      <height>51</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Цвет фона</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonBlack_2">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(0,0,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonWhite_2">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(255,255,255);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonYellow_2">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(255,255,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonRed_2">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(255,0,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonOrange_2">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(255,104,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonPurple_2">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(128,0,128);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonBlue_2">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(0,0,255);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonGreen_2">
       <property name="styleSheet">
        <string notr="true">QPushButton{
	background-color: rgb(0,255,0);
}</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>411</width>
      <height>401</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>520</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>(400;0)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>520</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>(400;400)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>100</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>(0;400)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>100</y>
      <width>31</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>(0;0)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ClearButton">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>100</y>
      <width>221</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Очистить рисунок</string>
    </property>
   </widget>
   <widget class="QPushButton" name="DrawButton">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>150</y>
      <width>221</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Построить окружность</string>
    </property>
   </widget>
   <widget class="QPushButton" name="AnalyzeEffButton">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>550</y>
      <width>301</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Временные характеристики для построения окружности</string>
    </property>
   </widget>
   <widget class="QPushButton" name="AnalyzeButton">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>390</y>
      <width>221</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Построить спектр окружностей</string>
    </property>
   </widget>
   <widget class="QPushButton" name="DrawEllipse">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>270</y>
      <width>221</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Построить эллипс</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_6">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>310</y>
      <width>221</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Координаты центра эллипса</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="XcE">
       <property name="placeholderText">
        <string>Координата X центра</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="YcE">
       <property name="placeholderText">
        <string>Координата Y центра</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_7">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>310</y>
      <width>221</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_6">
     <item>
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Полуоси</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="Ra">
       <property name="placeholderText">
        <string>Полуось a</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="Rb">
       <property name="placeholderText">
        <string>полуось b</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>590</y>
      <width>301</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Временные характеристики для построения эллипса</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>510</y>
      <width>221</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Построить спектр эллипсов</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>190</y>
      <width>221</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Координаты центра окружности</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="Xc">
       <property name="placeholderText">
        <string>Координата X центра</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="Yc">
       <property name="placeholderText">
        <string>Координата Y центра</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>420</y>
      <width>219</width>
      <height>81</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Координаты центра окружности</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="Xc_2">
       <property name="placeholderText">
        <string>Координата X центра</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="Yc_2">
       <property name="placeholderText">
        <string>Координата Y центра</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_9">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>550</y>
      <width>219</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_8">
     <item>
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Координаты центра эллипса</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="Xc_3">
       <property name="placeholderText">
        <string>Координата X центра</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="Yc_3">
       <property name="placeholderText">
        <string>Координата Y центра</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLineEdit" name="RBeg_2">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>571</y>
      <width>129</width>
      <height>21</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Начальное A</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="REnd_2">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>602</y>
      <width>129</width>
      <height>20</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Начальное B</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="RStep_2">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>540</y>
      <width>129</width>
      <height>20</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Шаг изменения A</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="RStep_3">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>601</y>
      <width>129</width>
      <height>21</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Шаг изменения B</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Circlenum_2">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>571</y>
      <width>129</width>
      <height>21</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Кол-во эллипсов</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="RBeg">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>450</y>
      <width>129</width>
      <height>22</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Начальный радиус</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="REnd">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>480</y>
      <width>129</width>
      <height>22</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Конечный радиус</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="RStep">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>450</y>
      <width>129</width>
      <height>22</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Шаг изменения радиуса</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Circlenum">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>480</y>
      <width>129</width>
      <height>22</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Кол-во окружностей</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>190</y>
      <width>219</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Радиус</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Rad">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>212</y>
      <width>219</width>
      <height>20</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Радиус окружности</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>959</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
